src/ripple/app/misc/ValidatorList.h

```diff
@@ -20,102 +20,233 @@
+    /// Determines whether a node is included on any recommended validator lists
     bool
-    removePermanentKey (
-        PublicKey const& identity);
+    listed (
+        PublicKey const& identity) const;
 
-    /** The number of installed permanent and ephemeral keys */
-    std::size_t
-    size () const;
+    /// Determine whether a validator list publisher key is trusted
+    bool
+    trustedPublisher (
+        PublicKey const& identity) const;
 
-    /** Invokes the callback once for every node in the UNL.
+    /** Invokes the callback once for every listed validator.
         @note You are not allowed to insert or remove any
-              nodes in the UNL from within the callback.
+              validators from within the callback.
 
         The arguments passed into the lambda are:
             - The public key of the validator;
-            - A (possibly empty) comment.
             - A boolean indicating whether this is a
-              permanent or ephemeral key;
+              trusted key;
     */
     void
-    for_each (
-        std::function<void(PublicKey const&, std::string const&, bool)> func) const;
+    for_each_listed (
+        std::function<void(PublicKey const&, bool)> func) const;
 
-    /** Load the list of trusted validators.
+    /** Load the list of trusted validators and validator list sites and keys.
 
-        The section contains entries consisting of a base58
+        The list of validators contains entries consisting of a base58
         encoded validator public key, optionally followed by
         a comment.
 
+        The list of validator list sites contains URIs to fetch recommended
+        validators.
+
+        The list of validator list keys contains base58 encoded account keys
+        used by trusted publishers to sign lists of recommended validators.
+
         @return false if an entry could not be parsed or
                 contained an invalid validator public key,
                 true otherwise.
     */
     bool
     load (
-        Section const& validators);
+        PublicKey const& localSigningKey,
+        std::vector<std::string> const& configValidators,
+        std::vector<std::string> const& validatorListSites,
+        std::vector<std::string> const& validatorListKeys,
+        std::vector<std::string> const& configManifest);
+
+    /** Update trusted validator list
+
+        Reset the list based on latest lists from validator sites, known active
+        validators, and stored validator manifests.
+    */
+    void
+    update (
+        ValidationSet const& activeValidators);
+
+    bool
+    removeList (PublicKey const& publisherKey);
+
+    int
+    quorum () const
+    {
+        return quorum_;
+    };
+
+    int
+    getFetchedSitesCount () const
+    {
+        return fetchedSites_;
+    }
+
+    /// Determines the quorum for the trusted validator list size.
+    static int
+    calcQuorum (
+        std::uint32_t const& nTrusted,
```

Comment:
Passing an `std::uint32_t` by `const&` is overkill. A plain-old `std::uint32_t` will do.